### ISSUE TITLE
Broke everything after rename current user

### ISSUE THREAD

**Issue by pplanel** [NixOS/nix#6413]
**Describe the bug**

After fixing a typo on my current user, rebuilding and renaming my /home/${user} to the correct name broke my nixos. Cant install packages with nix-env, my profile seems not being created, home-manager is refering to the old path.

**Steps To Reproduce**

1. rename current user in name and the variable
2. `sudo nixos-rebuild switch`
3. restart
4. broken system

**Expected behavior**

I was expecting that since nixos is handling the user mgmt it would fix the paths and sort things out.

nix-env (Nix) 2.5.0pre20211206_d1aaa7e

**Comment by thufschmitt** [comment]
I’m not sure there’s an easy way for Nix to fix that automatically. The problem (I think) is that on the first run, `nix-env` (and afair `home-manager`) creates some symlinks pointing to `/nix/var/nix/profiles/per-user/$USER`, but these probably didn’t get updated when you migrated your username.
You should try removing `~/.nix-profile` and `~/.nix-defexpr/*` (and maybe move `/nix/var/nix/profiles/per-user/$OLD_USER` to `/nix/var/nix/profiles/per-user/$NEW_USER` if you want to keep your old stuff). Hopefully that will be enough to fix your issue

**Comment by pplanel** [comment]
Thanks for taking time to help, it was a really unpleasant surprise, but i found my way around it. Your suggestions are spot on but as it wasnt enought at the same time I enabled `experimental-features` to play with flakes; so `nix-env` was out of game, `home-manager` trying to make sense of what happened. Anyways, I had to create a new user, moved manually my stuff to new home and installed `home-manager` again.
